### PR TITLE
[Soft Delete] Added --trash to delete cmd, added restore cmd

### PR DESCRIPTION
**Pull request by cscharf**
## Overview
Added soft-delete, trash and restore operations to the CLI to ensure the CLI can participate in and perform all of the necessary functions supported by Bitwarden (functionally complete). Also updated the CLI's `config` option to be able to more granularly provide custom environment URLs similar to how other clients do (like desktop and browser extension).

## Updates
- **bw.ts**: lint error correction (line length)
- **config.command.ts**: handle individual environment URL settings with null fallback (default) since commander will leave properties `undefined` vs. `null` if not passed
- **delete.command.ts**: handle the `-t, --trash` flag when deleting an item to instead soft-delete that item (move to trash) vs. permanently deleting it. Only supports the Object type, _item_
- **edit.command.ts**: prevent editing a deleted item with an error message about the `restore item <id>` command which must be run first
- **list.command.ts**: like the other clients, deleted items are not included in search/list by default, only now with the `--trash` flag passed in will it search, list, etc. the trash bin for list operations against _item_ Object types. For all other Object types, the `--trash` flag is ignored _(should this return an error?)_
- **restore.command.ts**: new command type for restoring deleted items, staged to be consistent as delete, edit, etc. so other types could presumably be added in the future
- **program.ts**: options and help text generator configuration for commander added for the new command and command options, examples, etc.

### CODE REVIEW

File: src/commands/delete.command.ts [hunk 1]
```diff
@@ -32,14 +32,18 @@ export class DeleteCommand {
...
             return Response.notFound();
         }
 
         try {
-            await this.cipherService.deleteWithServer(id);
+            if (cmd.trash) {
+                await this.cipherService.softDeleteWithServer(id);
+            } else {
+                await this.cipherService.deleteWithServer(id);
+            }
```
Comment (kspearrin) [inline src/commands/delete.command.ts:46]: Should it be the other way around? Where a regular delete does a soft delete?

File: src/commands/list.command.ts [hunk 2]
```diff
@@ -100,10 +103,12 @@ export class ListCommand {
                 }
                 return false;
             });
+        } else if (cmd.search == null || cmd.search.trim() === '') {
+            ciphers = ciphers.filter((c) => (cmd.trash || false) === c.isDeleted);
```
Comment (kspearrin) [inline src/commands/list.command.ts:107]: You could set `cmd.trash = cmd.trash || false`  at the beginning of this command and not have to evaluate that each time.

File: src/commands/delete.command.ts (hunk 1 above)
Comment (cscharf) [reply src/commands/delete.command.ts:46]: I didn't want to really mess with the current/default CLI behavior where there may be some automation or other things scripted for it already. But I can certainly reverse it to match the way the other clients work.

Comment (kspearrin) [reply src/commands/delete.command.ts:46]: That's probably a good idea. Let's leave it as-is.

Comment (cscharf) [reply src/commands/delete.command.ts]: I swapped it, good call. Also changed the command option switch key/shortcode and logic, tested and looks good.

**Pull request merged**